src/extension/importer/core/src/main/java/org/geoserver/importer/SpatialFile.java

```diff
@@ -99,36 +101,42 @@ public String apply(@Nullable StyleHandler input) {
 
         // getBaseName only gets the LAST extension so beware for .shp.aux.xml stuff
         final String baseName = getBaseName(file.getName());
+        final String baseExtension = getExtension(file.getName());
+        final String basePath = file.getParent();
+
+        // look for style files
+        Iterator styleExtensionsIt = styleExtensions.iterator();
+        while (styleFile == null && styleExtensionsIt.hasNext()) {
+            Object ext = styleExtensionsIt.next();
+            File style = new File(basePath, baseName + "." + ext);
+            if (style.exists()) {
+                // TODO: deal with multiple style files? for now we just grab the first
+                styleFile = style;
+            }
+        }
 
-        File[] files = file.getParentFile().listFiles();
-        if (files != null) {
-            for (File f : files) {
-                if (f.equals(file)) {
-                    continue;
-                }
-
-                if (!f.getName().startsWith(baseName)) {
-                    continue;
-                }
-
-                if (!f.isFile()) {
-                    continue;
-                }
-
-                String ext = f.getName().substring(baseName.length());
-                // once the basename is stripped, extension(s) should be present
-                if (ext.charAt(0) == '.') {
-                    if (".prj".equalsIgnoreCase(ext)) {
-                        prjFile = f;
-                    } else if (styleFile == null && styleExtensions.contains(ext.substring(1))) {
-                        // TODO: deal with multiple style files? for now we just grab the first
-                        styleFile = f;
+        // Looking for supplemental files
```

Comment:
This actually changes the behavior, some files that were caught before will not be in the list anymore. However, in fact getSuppFiles is not used for anything active in the import, it's just reflected out in the generated JSON when using the REST API, but not used to drive actual import, or to move files, or anything like that. So in the end, pretty invisible to the user, and not particularly useful (one may even just stop collecting them...).

Reply:
Right